src/GafferImage/Offset.cpp

```diff
@@ -167,8 +167,12 @@ IECore::ConstFloatVectorDataPtr Offset::computeChannelData( const std::string &c
 	}
 	else
 	{
+		Box2i inDataWindow = inPlug()->dataWindowPlug()->getValue();
```

Comment:
I feel like I'm missing something here. If the output data window is derived from the input data window, and downstream nodes must only pull tiles from the output data window, why do we need to do any guarding? As long as the downstream nodes are keeping their promise, doesn't that mean we are too?

If there is a genuine bug here though, then `hashChannelData()` would need updating to reflect the changes in `computeChannelData()`.

Reply:
We previously took the bound of the entire output tile, and shifted it into input space.  The output could have as little as one pixel overlapping with the actual output data window.  The tiles are aligned differently in input space, so of the 4 input tiles necessary to compute the output tile, it's possible that only one is valid.  I will add the update to hashChannelData.

Reply:
Thanks for the explanation.